tofu/harvester/modules/vm/main.tf

```diff
@@ -119,7 +105,20 @@ resource "harvester_virtualmachine" "vm" {
   }
 
   cloudinit {
-  user_data_secret_name = "${var.hostname_prefix}-secret-${random_string.random_suffix.result}"
+  user_data_secret_name = "${var.generate_name}-secret-${random_string.random_suffix.result}"
   network_data = ""
   }
 }
+
+resource "ansible_host" "node" {
+  for_each = { for node in local.node_names : node.name => node }
+  name = each.value.name
+  variables = {
+    # Connection vars.
+    ansible_user = var.ssh_user
+    ansible_host = harvester_virtualmachine.vm[each.key].network_interface[0].ip_address
+    ansible_role = join(",", each.value.role)
+
+  }
+  depends_on = [harvester_virtualmachine.vm]
+}
```

Comment:
If ansible is not run directly after this, does having this present any issues? Also, could I run the ansible if I didn't start with this tofu module? It might be helpful to add a comment here explaining.

Reply:
1. no, you can use the resources created from main.tf in whatever capacity you want after this. My understanding is, this is just setting up something that ansible can access, but doesn't break anything if you don't use ansible / these vars.
2. right now, no. This would need to be decoupled in some way in order to get ansible to run independently of terraform. 

again, this is modeling behavior from tofu/aws/modules/cluster_nodes

Reply:
We talked about the coupling of ansible and tofu offline and decided to decouple as a separate effort. Thank you!